src/bin/genemichaels.rs

```diff
@@ -199,7 +237,104 @@ fn main() {
             Offable::On(_) => true,
         },
     };
-    if args.files.is_empty() {
+    if args.workspace || args.folder {
+        // can use the same code for both
+        let res = || -> Result<()> {
+            {
+                let inst = time::Instant::now();
+                eprintln!("\x1B[1;32m  Formatting\x1B[0;22m {}...", if args.folder {
+                    "folder"
+                } else {
+                    "workspace"
+                });
+                let mut workspace_cargo_toml = None;
+                let c_dir = current_dir()?;
+                if args.workspace {
+                    let mut at: Option<&Path> = Some(&c_dir);
+                    while let Some(d) = at.take() {
+                        if d.join("Cargo.toml").exists() &&
+                            cargo_manifest::Manifest::from_path(d.join("Cargo.toml"))
+                                .ok()
+                                .and_then(|m| m.workspace)
+                                .is_some() {
+                            workspace_cargo_toml = Some(d.join("Cargo.toml"));
+                            break;
+                        }
+                        at = d.parent();
+                    }
+                }
+                let (dirs, working_folder) = match workspace_cargo_toml {
+                    Some(ct) => {
+                        // if workspace exists
+                        let path = ct.canonicalize()?;
+                        let workspace_folder = path.parent().unwrap();
+                        let manifest = cargo_manifest::Manifest::from_path(&path)?;
+                        match manifest.workspace.map(|wkspc| wkspc.members) {
+                            Some(mut mb) => {
+                                mb.push(String::new());
+                                (mb, workspace_folder.to_path_buf())
```

Comment:
Since the dirs are joined with the workspace folder, I think that could be done right away which would reduce the number of values that need to be passed around (i.e. instead of both `dirs` and `workspace_folder` it could just be a list of usable paths).